FirebaseAppCheck/Sources/AppAttestProvider/Storage/FIRAppAttestKeyIDStorage.m

```diff
@@ -36,18 +47,46 @@ - (instancetype)initWithAppName:(NSString *)appName appID:(NSString *)appID {
   if (self) {
     _appName = [appName copy];
     _appID = [appID copy];
+    _userDefaults = [[NSUserDefaults alloc] initWithSuiteName:kFIRAppAttestKeyIDStorageDefaultsSuiteName];
   }
   return self;
 }
 
+- (nonnull FBLPromise<NSString *> *)setAppAttestKeyID:(nullable NSString *)keyID {
+  [self storeAppAttestKeyID:keyID];
+  return [FBLPromise resolvedWith:keyID];
+}
+
 - (nonnull FBLPromise<NSString *> *)getAppAttestKeyID {
-  // TODO: Implement.
-  return [FBLPromise resolvedWith:nil];
+  NSString *appAttestKeyID = [self appAttestKeyIDFromStorage];
+  if (appAttestKeyID) {
+    return [FBLPromise resolvedWith:appAttestKeyID];
+  } else {
+    NSError *error = [FIRAppCheckErrorUtil appAttestKeyIDNotFound];
+    FBLPromise *rejectedPromise = [FBLPromise pendingPromise];
+    [rejectedPromise reject:error];
+    return rejectedPromise;
+  }
 }
 
-- (nonnull FBLPromise<NSString *> *)setAppAttestKeyID:(nullable NSString *)keyID {
-  // TODO: Implement.
-  return [FBLPromise resolvedWith:nil];
+#pragma mark - Helpers
+
+- (void)storeAppAttestKeyID:(nullable NSString *)keyID {
+  @synchronized (self.userDefaults) {
```

Comment:
optional: Alternatively we may use a private dispatch queue for synchronization  which is a preferable way when we have async API. You may leverage API like `+[FBLPromise onQueue:do:]` to organize to code.

Reply:
Discussed offline. Decision is to allow `FIRAppAttestKeyIDStorage`'s sole intended consumer, `FIRAppAttestProvider`, to manage synchronization of `FIRAppAttestKeyIDStorage` on its (`FIRAppAttestProvider`'s) internal serial queue.

A comment of the intended use has been added at the public interface definition of `FIRAppAttestKeyIDStorage`.